linetools/spectra/xspectrum1d.py

```diff
@@ -182,6 +192,15 @@ def __init__(self, wave, flux, sig=None, co=None, units=None, select=0,
             if wave.shape[0] != sig.shape[0]:
                 raise IOError("Shape of wave and sig vectors must be identical")
             self.data['sig'] = np.reshape(sig, (self.nspec, self.npix))
+            if masking != 'None':
+                for kk in range(self.nspec):
+                    sigval = np.where(self.data['sig'][kk] > 0.)[0]
+                    for key in self.data.dtype.names:
+                        if masking == 'edges':
+                            self.data[key][kk][0:sigval[0]].mask = True
+                            self.data[key][kk][sigval[-1]+1:].mask = True
+                        elif masking == 'all':
+                            self.data[key][kk][sigval].mask = True
```

Comment:
Yikes!  Good catch. 

On June 20, 2016 8:26:50 AM PDT, Nicolas Tejos <notifications@github.com> wrote:
>> @@ -182,6 +192,15 @@ def __init__(self, wave, flux, sig=None,
>co=None, units=None, select=0,
>>              if wave.shape[0] != sig.shape[0]:
>>                  raise IOError("Shape of wave and sig vectors must be
>identical")
>>              self.data['sig'] = np.reshape(sig, (self.nspec,
>self.npix))
>> +            if masking != 'None':
>> +                for kk in range(self.nspec):
>> +                    sigval = np.where(self.data['sig'][kk] > 0.)[0]
>> +                    for key in self.data.dtype.names:
>> +                        if masking == 'edges':
>> +                            self.data[key][kk][0:sigval[0]].mask =
>True
>> +                            self.data[key][kk][sigval[-1]+1:].mask =
>True
>> +                        elif masking == 'all':
>> +                            self.data[key][kk][sigval].mask = True
>
>This approach will mask out the good pixels and leave in the bad ones.
>'True' -> 'False'
>
>---
>You are receiving this because you modified the open/close state.
>Reply to this email directly or view it on GitHub:
>https://github.com/linetools/linetools/pull/226/files/cd9c6988891be9cf2fd10d75db2642452c237954#r67709919
>
>!DSPAM:826,57680b3c29679473325066!

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.